src/blog_vi/translations/engine.py

```diff
@@ -7,38 +9,66 @@ def __init__(self, landing: Landing, source_abbreviation: str):
         self.source_abbreviation = source_abbreviation
 
         self.settings = landing.settings
+        self.translator = self.get_translate_engine(self.settings)
 
-        # TODO:
-        #   - Validate settings
-        #   - Get translate provider from registry
+    def get_translate_engine(self, settings):
+        translator_cls = translation_provider_registry.get_provider(settings.translator)
+        return translator_cls.from_settings(settings)
```

Comment:
This should raise an error if engine not found

Reply:
`get_provider` has try except and it raise error if engine not found, or you mean not it

Reply:
`translator_cls` can be DeeplTranslator/GoogleTranslator or `get_provider` function raise error

Reply:
get provider shouldn't raise exceptions, as its context is not about it. This method should raise them

Reply:
if get provider is none -> raise an error